### PR TITLE
Diamond Inheritance Problem Issue 1459

### PR DESCRIPTION
**Pull request by Shatakshi-Choudhary**
## Related Issue
- Diamond Problem in Inheritance #1459 

## Proposed Changes
- Solved the Diamond Problem in Inheritance
- Added the code for the solution to the Diamond Problem

## Additional Info
- The diamond problem in Inheritance is solved by using the "virtual" keyword. We will make the two-child classes i.e B & C (who inherit from the same parent class i.e A) into virtual classes in order to avoid two copies of the parent class(class A) in the child class.

## Checklist

- [x] ✅ My code follows the code style of this project.
- [x] 📝 My change requires a change to the README.
- [x] 🎀 I have updated the README accordingly.
- [x] 👀 I have read the [**CONTRIBUTING**](https://github.com/Lakhankumawat/LearnCPP/blob/main/Contributing.md) document.
- [x] ✨ I have added tests to cover my changes.
- [x] 🚩 All new and existing tests passed.
- [x] 🌟 ed the repo

My code consists of the solution to the problem in multiple inheritance i.e known as Diamond Problem.

The Parent class(class A) constructor is called twice: once when the B class object is created and next when the C class object is created. The properties of the A class are inherited twice, giving rise to ambiguity.

My code fixes this issue #1459 

Tested the code by creating object of the sub-child class

- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Output Screenshots
|             Screenshot #1                |                  Screenshot #2  |
| The Diamond Problem  |                  Solved Problem     |
| Diamond1 |    Diamond2 |

### CODE REVIEW

**Review by Lakhankumawat — CHANGES_REQUESTED** [review]
You need to have only one cpp file with one README changes

**Comment by Lakhankumawat** [comment]
image


You need to follow particular guidelines for your README , please follow
these guidelines [click here](https://github.com/Lakhankumawat/LearnCPP#to-add-programs-using-stl-follow-same-approach-as--algorithms) 
also you can see how your README should look like [see here](https://github.com/Lakhankumawat/LearnCPP/blob/main/S-SortingAlgorithms/README.md) 
for more please go through these [here](https://youtu.be/7UgC3HDETRA)

File: O-OOPS/virtualFunctions.cpp [hunk 1]
```diff
@@ -0,0 +1,50 @@
+//                                                           LearnCPP
+//                                              VIRTUAL FUNCTIONS IN C++ #1237
+
```
Comment (Mrjoy832) [inline O-OOPS/virtualFunctions.cpp:3]: Why this is changed?

**Review by Mrjoy832 — CHANGES_REQUESTED** [review]

File: O-OOPS/virtualFunctions.cpp (hunk 1 above)
Comment (Shatakshi-Choudhary) [reply O-OOPS/virtualFunctions.cpp:3]: This issue was already created by someone. That's why I deleted my issue.

File: O-OOPS/README.md [hunk 2]
```diff
@@ -1,83 +1,168 @@
...
-* A binary Operator returns the same type as its parameters
-
-##### Example
-``int x = 2;``<br>
-``int y = 3;``<br>
-``int z = x + y;``
-
-<br>
-<hr>
+
```
Comment (Mrjoy832) [inline O-OOPS/README.md:1]: Your readme is incorrect 


You need to follow particular guidelines for your README , please follow
these guidelines [click here](https://github.com/Lakhankumawat/LearnCPP#to-add-programs-using-stl-follow-same-approach-as--algorithms) 
also you can see how your README should look like [see here](https://github.com/Lakhankumawat/LearnCPP/blob/main/S-SortingAlgorithms/README.md) 
for more please go through these [here](https://youtu.be/7UgC3HDETRA)

**Review by Mrjoy832 — CHANGES_REQUESTED** [review]

**Comment by Mrjoy832** [comment]
Star the repo and put tick on **Starred repo**

File: O-OOPS/README.md [hunk 3]
```diff
@@ -1,83 +1,132 @@
...
```
Comment (Mrjoy832) [inline O-OOPS/README.md:1]: - problem statement
- example
- images
- algo/approach
- time & space complexity
- advantages & disadvantages

**Review by Mrjoy832 — CHANGES_REQUESTED** [review]

**Review by Mrjoy832 — APPROVED** [review]

**Comment by Mrjoy832** [comment]
## LGTM🌟

**Review by Lakhankumawat — APPROVED** [review]
## LGTM ✨🎉

**Comment by Shatakshi-Choudhary** [comment]
Thank You!!

**Pull request merged**